doc/source/ray-core/examples/batch_prediction.ipynb

```diff
@@ -0,0 +1,321 @@
+    "\n",
+    "First, we define a callable class that’s structured with an interface (i.e. constructor) to load/cache the model, and the other to take in a file and perform prediction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import pyarrow.parquet as pq\n",
+    "import ray\n",
+    "\n",
+    "@ray.remote\n",
+    "class BatchPredictor:\n",
+    "    def __init__(self, model):\n",
+    "        self.model = model\n",
+    "        \n",
+    "    def predict(self, shard_path):\n",
+    "        df = pq.read_table(shard_path).to_pandas()\n",
+    "        result =self.model(df)\n",
+    "\n",
+    "        # Write out the prediction result.\n",
+    "        # Note: unless the driver will have to further process the\n",
+    "        # result (other than simply writing out to storage system),\n",
+    "        # writing out at remote task is recommended, as it can avoid\n",
+    "        # congesting or overloading the driver.\n",
+    "        # ...\n",
+    "\n",
+    "        # Here we just return the size about the result in this example.\n",
+    "        return len(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The constructor is called only once per actor worker. We use ActorPool to manage a set of actors that can receive prediction requests."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.util.actor_pool import ActorPool\n",
+    "\n",
+    "model = load_model()\n",
+    "model_ref = ray.put(model)\n",
+    "num_actors = 4\n",
+    "actors = [BatchPredictor.remote(model_ref) for _ in range(num_actors)]\n",
+    "pool = ActorPool(actors)\n",
+    "input_files = [\n",
+    "        f\"s3://anonymous@air-example-data/ursa-labs-taxi-data/downsampled_2009_full_year_data.parquet\"\n",
+    "        f\"/fe41422b01c04169af2a65a83b753e0f_{i:06d}.parquet\"\n",
+    "        for i in range(12)\n",
+    "]\n",
+    "for file in input_files:\n",
+    "    pool.submit(lambda a, v: a.predict.remote(v), file)\n",
+    "while pool.has_next():\n",
+    "    print(\"Prediction output size:\", pool.get_next())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Batch prediction with GPUs\n",
+    "\n",
+    "If your cluster has GPU nodes and your predictor can utilize the GPUs, you can direct the tasks or actors to those GPU nodes by specifying num_gpus. Ray will schedule them onto GPU nodes accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@ray.remote # NOTE: Replace with @ray.remote(num_gpus=1) to use GPUs\n",
+    "def make_prediction(model, shard_path):\n",
+    "    df = pq.read_table(shard_path).to_pandas()\n",
+    "    result = model(df)\n",
```

Comment:
You may also want to leave a tip here that shows how one might move their model to the GPU.

For example in TF and Torch the user will need to manually do this.

Reply:
Good point, done.